### PR TITLE
Implemented showing the number of preauthorised commands left. #1035

### PR DESCRIPTION
**Pull request by xhypeDE**
### Background
This change was inspired by user request in issue #1035, aiming to improve the user experience of Auto-GPT by providing clear information about the number of authorized commands left and the details of the next action. This enhancement is particularly useful for users who pre-authorize commands and want to track their progress.

### Changes
In this pull request, the following modifications were made to the `agent.py` file:

1. Added a conditional statement `elif` under the main control flow to handle cases where `self.next_action_count > 0` and `cfg.continuous_mode` is `False`.
2. Used the `logger.typewriter_log()` function to display the number of authorized commands left and the details of the next action, including the command name and its arguments.

### Documentation
The changes are documented through in-code comments and this external documentation, which clearly explain the new behavior added to Auto-GPT after merging this pull request. The number of authorized commands left will be displayed in cyan color when the user is not in continuous mode, and the details of the next action, including the command name and its arguments, will be displayed in both continuous and non-continuous modes.

### Test Plan
To test this functionality:

1. Ran Auto-GPT with the updated `autogpt/agent/agent.py` file .
2. Pre-authorized commands using `y -n`, where `n` is the numerical value of commands you want to authorize.
3. Observed the number of authorized commands left and the details of the next action being displayed after every message in the specified color when not in continuous mode.
4. Verified that the output aligns with the expected behavior as described.

Flake8 and PyTest run without any issues.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -195,6 +195,18 @@ def start_interaction_loop(self):
                 elif user_input == "EXIT":
                     print("Exiting...", flush=True)
                     break
+            elif self.next_action_count > 0 and not cfg.continuous_mode:
+                # Print authorized commands left value
+                logger.typewriter_log(
+                    f"{Fore.CYAN}PREAUTHORISED COMMANDS LEFT: {Style.RESET_ALL}{self.next_action_count}"
```
Comment (hdkiller) [inline autogpt/agent/agent.py]: "pre-authorized" or "Authorized commands left:" might be easier to read?

Comment (neoyubi) [reply autogpt/agent/agent.py]: I just implemented the suggested changes and filled out the PR properly. Please review. @hdkiller thank you!

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
This looks like a useful and straightforward code addition that could probably be considered for review/integration regardless of the ongoing re-arch effort ?

**Review by k-boikov — APPROVED** [review]

**Comment by xhypeDE** [comment]
Thanks to everyone involved in this PR. Appreciate the efforts!

**Comment by Boostrix** [comment]
> Thanks to everyone involved in this PR. Appreciate the efforts!

if you haven't already, please consider joining discord to collect your contributor/catalyst status there and help us get other PRs/issues implemented, reviewed and integrated.

**Pull request merged**